selfservice/flow/settings/handler_test.go

```diff
@@ -159,6 +159,24 @@ func TestHandler(t *testing.T) {
 				assert.Contains(t, res.Request.URL.String(), reg.Config(context.Background()).SelfServiceFlowLoginUI().String())
 				assert.EqualValues(t, "Please complete the second authentication challenge.", gjson.GetBytes(body, "ui.messages.0.text").String(), "%s", body)
 			})
+
+			t.Run("case=redirects with 303", func(t *testing.T) {
+				c := http.DefaultClient
+				// don't get the reference, instead copy the values, so we don't alter the client directly.
+				*c = *publicTS.Client()
+				// prevent the redirect
+				c.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+					return http.ErrUseLastResponse
+				}
+				req, err := http.NewRequest("GET", publicTS.URL+settings.RouteInitBrowserFlow, nil)
+				require.NoError(t, err)
+
+				res, err := c.Do(req)
+				require.NoError(t, err)
+				// here we check that the redirect status is 303
+				require.Equal(t, http.StatusSeeOther, res.StatusCode)
+				defer res.Body.Close()
+			})
 		})
```

Comment:
Not sure if this will work, since we need a session first before we initialize a settings flow

https://www.ory.sh/kratos/docs/self-service/flows/user-settings#user-and-profile-settings-for-server-side-browser-clients

Also shown in one of the tests above:

```go
t.Run("description=success", func(t *testing.T) {
	user1 := testhelpers.NewHTTPClientWithArbitrarySessionCookie(t, reg)
	res, body := initFlow(t, user1, false)
	assert.Contains(t, res.Request.URL.String(), reg.Config(context.Background()).SelfServiceFlowSettingsUI().String())
    assertion(t, body, false)
})
```

Reply:
The test ran through, I switched the default client with the one including the session cookie anyways since all other tests here use it as well

Reply:
I also instantiated a new http client instead of using `http.DefaultClient` because some subsequent tests where failing due to the changed redirect behavior on the default instance.